### PR TITLE
Update cuFINUFFT docs

### PR DESCRIPTION
**Pull request by janden**
New installation instructions, updated C/C++ API and updated Python API. Also added some getting started examples in the various frameworks.

### CODE REVIEW

File: docs/c_gpu.rst [hunk 1]
```diff
@@ -221,15 +221,15 @@ For type 1 these points are "sources", but for type 2, "targets".
...
 
     Returns:
 
     status      zero if success, otherwise an error occurred
 
-Note: The user must not change the contents of the GPU arrays ``x``, ``y``, or ``z`` (in 3D case) between this step and the below execution step. They are read in the execution step also.
+Note: The user must not change the contents of the GPU arrays ``x``, ``y``, or ``z`` between this step and the below execution step. They are read in the execution step also.
 
-Note: The actual plan (not its pointer is passed in); new GPU arrays are allocated and filled in the internal plan struct that the plan points to.
+Note: The actual plan (not its pointer) is passed in; new GPU arrays are allocated and filled in the internal plan struct that the plan points to.
```
Comment (ahbarnett) [inline docs/c_gpu.rst:232]: this sentence is cryptic. do we understand it?

File: docs/python_gpu.rst [hunk 2]
```diff
@@ -30,32 +27,22 @@ We then proceed to setting up a few parameters.
...
-    plan = cufinufft.Plan(1, (N,), dtype=np.float64)
-
-    # set the nonuniform points
-    plan.setpts(to_gpu(x))
+    x_gpu = 2 * cp.pi * cp.random.uniform(size=M)
+    c_gpu = (cp.random.standard_normal(size=M)
+             + 1J * cp.random.standard_normal(size=M))
 
-With everything set up, we are now ready to execute the plan.
+Now that the data is prepared, we can call cuFINUFFT to compute the transform of the ``M`` source point onto a grid of ``N`` modes
```
Comment (ahbarnett) [inline docs/python_gpu.rst:34]: point -> points
transform from the ...  not transform of the...

**Review by ahbarnett — APPROVED** [review]
Thanks! The corrections to docs are essential. The getting started examples are great, for these different py-gpu styles.   Happy for you to merge.  Libin may help with the missing CI.

**Comment by janden** [comment]
> The new borken CI appears to be pytorch, since you added all 4 py-GPU interface examples to the CI?

Yes that's odd. Was running fine when I was testing on the cluster. It also seems to fail in a way that should lead the other (non-torch) examples to fail as well. Will see if I can narrow it down.

File: docs/c_gpu.rst (hunk 1 above)
Comment (janden) [reply docs/c_gpu.rst:232]: No I was trying to salvage it. I think the first part is just contrasting with `makeplan` where we pass a pointer to a plan (handle) and here we're just passing the handle. The second part I don't know whether it's actually needed.

File: docs/python_gpu.rst (hunk 2 above)
Comment (janden) [reply docs/python_gpu.rst]: Done.

File: docs/c_gpu.rst (hunk 1 above)
Comment (janden) [reply docs/c_gpu.rst]: Took a stab at fixing this.

**Comment by janden** [comment]
Oddly seems to have been solved by reordering imports (see 94bcdd73d0a69177d2232229cfecc80468e28e9c). Doesn't make too much sense to me but will leave as is and make an issue.

**Comment by janden** [comment]
Going to merge this. Can make an issue if there are things to fix here.

**Pull request merged**